### PR TITLE
add resource group name rule

### PR DESCRIPTION
**Pull request by weiliu1031**
Signed-off-by: Wei Liu <wei.liu@zilliz.com>
Issue: #21484

### CODE REVIEW

**Comment by weiliu1031** [comment]
rerun ut

File: internal/proxy/util.go [hunk 1]
```diff
@@ -113,6 +113,34 @@ func validateCollectionNameOrAlias(entity, entityType string) error {
...
+func ValidateResourceGroupName(entity string) error {
+	if entity == "" {
+		return fmt.Errorf("resource group name %s should not be empty", entity)
+	}
+
+	invalidMsg := fmt.Sprintf("Invalid resource group name %s.", entity)
+	if int64(len(entity)) > Params.ProxyCfg.MaxNameLength {
+		msg := invalidMsg + "The length of a resource group name must be less than " +
+			strconv.FormatInt(Params.ProxyCfg.MaxNameLength, 10) + " characters."
+		return errors.New(msg)
```
Comment (congqixia) [inline internal/proxy/util.go:125]: Use `fmt.Errorf()` instead of string concats

File: internal/proxy/util.go [hunk 2]
```diff
@@ -113,6 +113,34 @@ func validateCollectionNameOrAlias(entity, entityType string) error {
...
+	}
+
+	firstChar := entity[0]
+	if firstChar != '_' && !isAlpha(firstChar) {
+		msg := invalidMsg + "The first character of a resource group name must be an underscore or letter."
+		return errors.New(msg)
```
Comment (congqixia) [inline internal/proxy/util.go:131]: Ditto

File: internal/proxy/impl.go [hunk 3]
```diff
@@ -5060,9 +5070,26 @@ func (node *Proxy) DescribeResourceGroup(ctx context.Context, request *milvuspb.
...
+				Reason:    err.Error(),
+			},
+		}
+	}
+	if err := ValidateResourceGroupName(request.GetResourceGroup()); err != nil {
+		log.Warn("DescribeResourceGroup failed",
+			zap.Error(err),
+		)
+		return GetErrResponse(err), nil
+	}
```
Comment (congqixia) [inline internal/proxy/impl.go:5089]: Fells weird that got an error state bad rg name instead of rg not exists

**Comment by weiliu1031** [comment]
rerun ut

**Comment by weiliu1031** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**